File: Array/283.move-zeroes.js
var moveZeroes = array => {
    let pointer = 0;
    for (let i = 0; i < array.length; i++) {
        if (array[i] !== 0) {
            let temp = array[i];
            array[i] = array[pointer];
            array[0] = temp;
            pointer++;
        }
    }
    return array;
}

File: Array/283.move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int zeros = 0;
        for(int i = 0; i<nums.size(); i++){
            if(nums[i]==0){
                zeros++; 
                nums.erase(nums.begin() + i);
            }
        }
        for(int j=0; j<zeros; j++){
            nums.push_back(0);
        }
    }
};